mm2src/coins/z_coin/storage/walletdb/mod.rs

```diff
@@ -40,6 +40,7 @@ impl<'a> WalletDbShared {
         zcoin_builder: &ZCoinBuilder<'a>,
         checkpoint_block: Option<CheckPointBlockInfo>,
         z_spending_key: &ExtendedSpendingKey,
+        sync_params: Option<SyncStartPoint>,
```

Comment:
We can pass here a bool instead of the whole `sync_params`, maybe call it `continue_prev_sync` and calculate the value and pass it from the `init_light_client` fn
```rust
let min_height = blocks_db.get_earliest_block().await?;
// check if no sync_params was provided and continue syncing from last height in db if it's > 0.
let continue_prev_sync = min_height > 0 && sync_params.is_none();
```
overall consider the following changes
```diff
Index: mm2src/coins/z_coin/z_rpc.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mm2src/coins/z_coin/z_rpc.rs b/mm2src/coins/z_coin/z_rpc.rs
--- a/mm2src/coins/z_coin/z_rpc.rs	(revision 5f32cf5d8fd09711b44376401ddebfe540a19af1)
+++ b/mm2src/coins/z_coin/z_rpc.rs	(date 1694531931787)
@@ -365,7 +365,7 @@
     consensus_params: ZcoinConsensusParams,
     checkpoint_block: Option<CheckPointBlockInfo>,
     evk: ExtendedFullViewingKey,
-    sync_params: Option<SyncStartPoint>,
+    continue_prev_sync: bool,
 ) -> Result<WalletDb<ZcoinConsensusParams>, MmError<ZcoinClientInitError>> {
     async_blocking({
         move || -> Result<WalletDb<ZcoinConsensusParams>, MmError<ZcoinClientInitError>> {
@@ -379,15 +379,13 @@
                 .map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
             init_wallet_db(&db).map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
 
-            // check if no sync_params was provided and continue syncing from last height in db if it's > 0.
-            let active_sync_block = min_sync_height.unwrap_or(0) > 0 && sync_params.is_none();
             // Check if the initial block height is less than the previous synchronization height and
             // Rewind walletdb to the minimum possible height.
             if db.get_extended_full_viewing_keys()?.is_empty()
-                || (!active_sync_block && init_block_height != min_sync_height)
+                || (!continue_prev_sync && init_block_height != min_sync_height)
             {
                 // let user know we're clearing cache and resyncing from new provided height.
-                if min_sync_height.unwrap_or(0) > 0 && sync_params.is_some() {
+                if min_sync_height.unwrap_or(0) > 0 {
                     info!("Older/Newer sync height detected!, rewinding walletdb to new height: {init_block_height:?}");
                 }
 
@@ -486,17 +484,17 @@
     let maybe_checkpoint_block = light_rpc_clients
         .checkpoint_block_from_height(sync_height.max(sapling_activation_height))
         .await?;
+    let min_height = blocks_db.get_earliest_block().await?;
+    // check if no sync_params was provided and continue syncing from last height in db if it's > 0.
+    let continue_prev_sync = min_height > 0 && sync_params.is_none();
 
-    let wallet_db = WalletDbShared::new(builder, maybe_checkpoint_block, z_spending_key, sync_params.clone())
+    let wallet_db = WalletDbShared::new(builder, maybe_checkpoint_block, z_spending_key, continue_prev_sync)
         .await
         .mm_err(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
 
-    let min_height = blocks_db.get_earliest_block().await?;
-    // check if no sync_params was provided and continue syncing from last height in db if it's > 0.
-    let active_sync_block = min_height > 0 && sync_params.is_none();
-    if !active_sync_block && (sync_height != min_height as u64) {
+    if !continue_prev_sync && (sync_height != min_height as u64) {
         // let user know we're clearing cache and resyncing from new provided height.
-        if min_height > 0 && sync_params.is_some() {
+        if min_height > 0 {
             info!("Older/Newer sync height detected!, rewinding blocks_db to new height: {sync_height:?}");
         }
         blocks_db
@@ -560,7 +558,7 @@
         is_pre_sapling: false,
         actual: checkpoint_height,
     };
-    let wallet_db = WalletDbShared::new(builder, checkpoint_block, z_spending_key, None)
+    let wallet_db = WalletDbShared::new(builder, checkpoint_block, z_spending_key, true)
         .await
         .mm_err(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
 
Index: mm2src/coins/z_coin/storage/walletdb/mod.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mm2src/coins/z_coin/storage/walletdb/mod.rs b/mm2src/coins/z_coin/storage/walletdb/mod.rs
--- a/mm2src/coins/z_coin/storage/walletdb/mod.rs	(revision 5f32cf5d8fd09711b44376401ddebfe540a19af1)
+++ b/mm2src/coins/z_coin/storage/walletdb/mod.rs	(date 1694531944527)
@@ -3,7 +3,7 @@
 use zcash_primitives::zip32::ExtendedSpendingKey;
 
 cfg_native!(
-    use crate::z_coin::{CheckPointBlockInfo, SyncStartPoint, ZcoinConsensusParams};
+    use crate::z_coin::{CheckPointBlockInfo, ZcoinConsensusParams};
     use crate::z_coin::z_rpc::create_wallet_db;
 
     use parking_lot::Mutex;
@@ -40,7 +40,7 @@
         zcoin_builder: &ZCoinBuilder<'a>,
         checkpoint_block: Option<CheckPointBlockInfo>,
         z_spending_key: &ExtendedSpendingKey,
-        sync_params: Option<SyncStartPoint>,
+        continue_prev_sync: bool,
     ) -> MmResult<Self, WalletDbError> {
         let wallet_db = create_wallet_db(
             zcoin_builder
@@ -49,7 +49,7 @@
             zcoin_builder.protocol_info.consensus_params.clone(),
             checkpoint_block,
             ExtendedFullViewingKey::from(z_spending_key),
-            sync_params,
+            continue_prev_sync,
         )
         .await
         .mm_err(WalletDbError::ZcoinClientInitError)?;
```

Reply:
thanks, done.